src/CoffeeShop.js

```diff
@@ -0,0 +1,66 @@
+function CoffeeShop (shopData) {
+  this.shopName = shopData.shopName;
```

Comment:
It's good that you've extracted this logic into an object, but you could have gone much further, and extracted into more than one object - the coffee shop is doing way too much and violates the single responsibility principle. Ideally it should just be a container that holds other modules that it delegates to which handle different parts of the app like discounts, calculations etc.